src/main/java/org/folio/bulkops/util/FqmContentFetcher.java

```diff
@@ -79,7 +84,22 @@ private InputStream task(UUID queryId, EntityType entityType, int chunk, int tot
     int limit = Math.min(chunkSize, total - offset);
     var response = queryClient.getQuery(queryId, offset, limit).getContent();
     return new ByteArrayInputStream(response.stream()
-        .map(json -> json.get(getContentJsonKey(entityType)).toString()).collect(Collectors.joining("\n"))
+        .map(json -> {
+          var tenant = json.get(getContentTenantKey(entityType));
+          if (tenant == null) {
+            tenant = folioExecutionContext.getTenantId();
+          }
+          var jsonb = json.get(getContentJsonKey(entityType));
+          try {
+            ObjectNode extendedRecordWrapper = objectMapper.createObjectNode();
+            extendedRecordWrapper.set("entity", objectMapper.readTree(jsonb.toString()));
+            extendedRecordWrapper.put("tenantId", tenant.toString());
+            return objectMapper.writeValueAsString(extendedRecordWrapper);
+          } catch (Exception e) {
+            log.error("Error processing JSON content for entity type {}: {}", entityType, e.getMessage());
+            throw new FqmFetcherException("Error processing JSON content", e);
+          }
+        }).collect(Collectors.joining("\n"))
```

Comment:
'\n' new line separator is already introduced as constant (or it makes sense to introduce such constant).